x/dsmr/node.go

```diff
@@ -217,31 +231,41 @@ func (n *Node[T]) BuildChunk(
 	return chunk, chunkCert, n.storage.AddLocalChunkWithCert(chunk, &chunkCert)
 }
 
-func (n *Node[T]) BuildBlock(parent Block, timestamp int64) (Block, error) {
+func (n *Node[T]) BuildBlock(ctx context.Context, parent Block, timestamp int64) (Block, error) {
 	if timestamp <= parent.Timestamp {
 		return Block{}, ErrTimestampNotMonotonicallyIncreasing
 	}
 
 	chunkCerts := n.storage.GatherChunkCerts()
+	oldestAllowed := max(0, timestamp-int64(n.validityWindowDuration))
+
+	emapChunkCert := make([]*emapChunkCertificate, len(chunkCerts))
+	for i := range emapChunkCert {
+		emapChunkCert[i] = &emapChunkCertificate{*chunkCerts[i]}
+	}
+	dup, err := n.validityWindow.IsRepeat(ctx, NewExecutionBlock(parent), emapChunkCert, oldestAllowed)
```

Comment:
nit: I had to inspect the implementation to realize this is a bitset... can we just call this `bitSet`, or `duplicates`? `dup` I assumed was a `bool` value at first...

Reply:
I'm ok changing this, just please note that the same applied to other placed in the codebase that are already using the `dup` variable name for the same purpose.